core/src/rpc.rs

```diff
@@ -1362,6 +1370,64 @@ impl RpcSol for RpcSolImpl {
         Ok(signature)
     }
 
+    fn simulate_transaction(
+        &self,
+        meta: Self::Metadata,
+        data: String,
+        config: Option<RpcSimulateTransactionConfig>,
+    ) -> RpcResponse<TransactionStatus> {
+        let config = config.unwrap_or(RpcSimulateTransactionConfig { sig_verify: false });
+
+        let data = bs58::decode(data).into_vec().unwrap();
+        if data.len() >= PACKET_DATA_SIZE {
+            info!(
+                "simulate_transaction: transaction too large: {} bytes (max: {} bytes)",
+                data.len(),
+                PACKET_DATA_SIZE
+            );
+            return Err(Error::invalid_request());
+        }
+        let transaction: Transaction = bincode::config()
+            .limit(PACKET_DATA_SIZE as u64)
+            .deserialize(&data)
+            .map_err(|err| {
+                info!("simulate_transaction: deserialize error: {:?}", err);
+                Error::invalid_request()
+            })?;
```

Comment:
Move this to a helper fn (like `verify_pubkey`), since it is duplicated in `send_transaction`? Will also help ensure that `simulate` is checking the same things as `send`

Reply:
Done (1a3b8cc7e2583fbd0aba69d409c47aed14e3ce36)